prac_08/box_layout_demo.py

```diff
@@ -0,0 +1,25 @@
+"""Modify Existing GUI Program"""
+
+from kivy.app import App
+from kivy.lang import Builder
+
+
+class BoxLayoutDemo(App):
+    """Main program - Kivy app to greet user based on text input."""
+    def build(self):
+        """Build the Kivy GUI."""
+        self.title = "Box Layout Demo"
+        self.root = Builder.load_file("box_layout_demo.kv")
+        return self.root
+
+    def handle_greet(self):
+        """Handle pressing greet button."""
+        self.root.ids.output_label.text = f"Hello {self.root.ids.input_name.text}"
+
+    def handle_clear(self):
+        """Handle pressing clear button."""
+        self.root.ids.output_label.text = ""
+        self.root.ids.input_name.text = ""
+
```

Comment:
There is no handling for cases where input_name or output_label might not exist. A try-except block could help avoid crashes in such cases

Reply:
True. I didn't add it because the question didn't ask for it.